apps/nestjs-wallet/src/vc-api/exchanges/dtos/custom-validators/presentation-definition-credential-query.validator.ts

```diff
@@ -0,0 +1,33 @@
+import { IPresentationDefinition, PEX, Status } from '@sphereon/pex';
+import { registerDecorator, ValidationOptions, ValidationArguments } from 'class-validator';
+
+/**
+ * https://github.com/typestack/class-validator#custom-validation-decorators
+ */
+export function IsPresentationDefinitionCredentialQuery(validationOptions?: ValidationOptions) {
+  return function (object: Object, propertyName: string) {
+    registerDecorator({
+      name: 'isPresentationDefinitionCredentialQuery',
+      target: object.constructor,
+      propertyName: propertyName,
+      constraints: [],
+      options: validationOptions,
+      validator: {
+        validate(value: IPresentationDefinition, args: ValidationArguments) {
+          const pex = new PEX();
+          try {
+            const validated = pex.validateDefinition(value);
+            const resultArray = Array.isArray(validated) ? validated : [validated];
+            const statuses = resultArray.map((checked) => checked.status);
+            if (statuses.includes(Status.ERROR) || statuses.includes(Status.WARN)) {
+              return false;
```

Comment:
What do you think about throwing here an error? With this approach you will have only one return statements in `catch` block. Also you have only one place for "invalidating" logic in your code

Reply:
Hmm, maybe I don't understand the custom validators for `class-validator` that well. But shouldn't a validator return true or false? I think that though `pex` might throw an error to indicate an invalid presentation definition, I think it makes sense for the return of the custom validator to reliably be true or false. Is that not how you see it?